newsroom/navigations/views.py

```diff
@@ -1,77 +1,123 @@
-    set_version_creator,
-)
+from newsroom.utils import query_resource
+from newsroom.core import get_current_wsgi_app
+from newsroom.flask import get_file_from_request
 from newsroom.assets import save_file_and_get_url
 
+from .service import NavigationsService
+from .utils import get_navigations_as_list
+
+
+navigations_endpoints = EndpointGroup("navigations", __name__)
+
+
+async def get_settings_data():
+    all_navigations = await get_navigations_as_list()
 
-def get_settings_data():
     return {
         "products": list(query_resource("products")),
-        "navigations": list(query_resource("navigations")),
+        "navigations": all_navigations,
         "sections": [
-            s for s in get_current_app().as_any().sections if s.get("_id") != "monitoring"
+            s for s in get_current_wsgi_app().sections if s.get("_id") != "monitoring"
         ],  # monitoring has no navigation
     }
 
 
-@blueprint.route("/navigations", methods=["GET"])
-def index():
-    navigations = list(query_resource("navigations", lookup=None))
-    return jsonify(navigations), 200
+class SearchParams(BaseModel):
+    q: str | None = None
+
+
+class RouteArguments(BaseModel):
+    id: str
+
 
+async def get_nav_data_from_request(request: Request) -> dict[str, Any]:
+    form_data = await request.get_form()
+    nav_data = form_data.get("navigation")
+    if nav_data is None:
+        await request.abort(400)
 
-@blueprint.route("/navigations/search", methods=["GET"])
-def search():
+    return json.loads(nav_data)
+
+
+@navigations_endpoints.endpoint("/navigations", methods=["GET"])
+async def index():
+    navigations = await get_navigations_as_list()
+    return Response(navigations)
+
+
+@navigations_endpoints.endpoint("/navigations/search", methods=["GET"])
+async def search(_a, params: SearchParams, _q):
     lookup = None
-    if request.args.get("q"):
-        regex = re.compile(".*{}.*".format(request.args.get("q")), re.IGNORECASE)
+    if params.q:
+        regex = re.compile(".*{}.*".format(params.q), re.IGNORECASE)
         lookup = {"name": regex}
-    products = list(query_resource("navigations", lookup=lookup))
-    return jsonify(products), 200
+    cursor = await NavigationsService().search(lookup)
+    navigations = await cursor.to_list_raw()
+    return Response(navigations)
 
 
-@blueprint.route("/navigations/new", methods=["POST"])
+@navigations_endpoints.endpoint("/navigations/new", methods=["POST"])
 @admin_only
-async def create():
-    data = json.loads((await request.form)["navigation"])
-    nav_data = await _get_navigation_data(data)
-    product_ids = nav_data.pop("products", [])
+async def create(request: Request):
+    nav_data = await get_nav_data_from_request(request)
+    service = NavigationsService()
 
-    set_original_creator(nav_data)
-    ids = get_resource_service("navigations").post([nav_data])
+    creation_data = await prepare_navigation_data(nav_data)
+    creation_data["id"] = service.generate_id()
```

Comment:
We could probably have this in the base async service create method. If supplied item is a dict and `_id` not defined, assign using the `self.generate_id()` method.

This way we dont' have to worry about it with these create item routes

Reply:
yeah, good idea. I'll do that in a separate PR to superdesk-core and then another one for newsroom core.